python/deltalake/table.py

```diff
@@ -4,6 +4,7 @@
 from typing import Any, List, Optional, Tuple
 from urllib.parse import urlparse
 
+import pandas as pd
```

Comment:
pandas is an optional dependency, this would break some of the downstream code, we need to wrap this import with try except.

Reply:
Mmm.. though you would still need it for the type checking (`def to_pandas() -> pd.DataFrame`) and in the unit tests. Would you have any suggestions how to incorporate that?

Reply:
that's a good call, have you tried writing it out as a string: `"pd.DataFrame"`? mypy should still be smart enough to pick it up.

Reply:
Tried that out, but `mypy` still complains because pandas is not available during the lint check. Added a `#ignore` to suppress it for now.